systemtest/src/main/java/io/strimzi/systemtest/Environment.java

```diff
@@ -370,16 +379,37 @@ private static <T> T getOrDefault(String var, Function<String, T> converter, T d
         return returnValue;
     }
 
-    private static JsonNode loadConfigurationFile() {
-        config = System.getenv().getOrDefault(CONFIG_FILE_PATH_ENV,
-                Paths.get(System.getProperty("user.dir"), "config.json").toAbsolutePath().toString());
-        ObjectMapper mapper = new ObjectMapper();
+    private static void saveConfigurationFile() throws IOException {
+        Path logPath = Path.of(TEST_LOG_DIR);
+        Files.createDirectories(logPath);
+
+        LinkedHashMap<String, String> toSave = new LinkedHashMap<>();
+
+        VALUES.forEach((key, value) -> {
+            if (isWriteable(key, value)) {
+                toSave.put(key, value);
+            }
+        });
+
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        mapper.writerWithDefaultPrettyPrinter().writeValue(logPath.resolve("config.yaml").toFile(), toSave);
```

Comment:
Maybe `config.yaml` can be constant either here or in `TestConstants`?

Reply:
I don't think it will be used somewhere else so I would keep it in string.